Add Jersey To List Of Countries/Flags

gillerg8 opened the pull request:
There was a requirement made by a client to add Jersey to the dropdown, not sure if this is something that you'd like to pull in. Let me know if there was anything I should I have done different to add this country. Thanks for creating this great component, it's awesome!!

bl00mber commented:
1. Since Jersey is not an independent country, [dependent territories](https://en.wikipedia.org/wiki/Dependent_territory) **with the low population count (~100.000** are preferred not to be included to the main list (although they are included currently). So I suggest you to add the 8th boolean parameter (Dependent territory) to the country (in this case, territory) model of the new file `rawTerritories.js`, and boolean parameter to the input (enableTerritories), which is false by default. If parameter enableTerritories, which should be passed as the 2nd argument with `new CountryData()` initialization, is true, during country list formation in `CountryData.js` these countries should be loaded from the new file `rawTerritories.js` within the CountryData constructor after `filterRegions()` call in the `initTerritories()` class method that I suggest you to write, which should return updated (with included territories (the alphabetical order needs to be preserved)) `filteredCountries` array.

2. I suggest you to move territories **with the low population count** from the `rawCountries.js` to the `rawTerritories.js`.

3. .png sprites needs to be [compressed](https://tinypng.com)

4. You can include flags also to `high-res.png` (or use dummy flag)

5. Dummy (as in the case of some countries which do not have their own flag in the `high-res.png`) or real flags of the added territories needs to be injected into `high-res-flags.less`.

6. Add enableTerritories property description to README.md under the prop enableAreaCodes.

gillerg8 commented:
I've made some updates and tested it out using the demo application, seems to be working as expected, but please let me know if you see anything off about the changes I've made.

1. Split out the dependent territories with a low population count from the rawCountries file into it's their own rawTerritories file by following this wikipedia article up until Jersey: https://en.wikipedia.org/wiki/List_of_countries_and_dependencies_by_population
2. Compressed the .png image using the site you've provided
3. Added the enableTerritories property description to README.md under the prop enableAreaCodes

bl00mber commented on README.md:
```diff
@@ -151,21 +156,28 @@ import 'react-phone-input-2/lib/style.css'
+    - [Check validity of the phone number](#check-validity-of-the-phone-number)
+    - [CDN](#cdn)
+  - [Contributing](#contributing)
+  - [License](#license)
```
Please revert to original contents, its structure was selected intentionally because the contents chapter is placed lower than first chapters, so there's no need to specify them.

gillerg8 replied:
Sorry about that, my editor did this without letting me know, should be all set now though!

bl00mber commented on src/CountryData.js:
```diff
@@ -77,15 +78,21 @@ function initCountries(prefix, defaultMask, alwaysDefaultMask) {
+        initCountriesAndAreaCodes(rawTerritories, enableAreaCodes, prefix, defaultMask, alwaysDefaultMask) :
+        initCountries(rawTerritories, prefix, defaultMask, alwaysDefaultMask);
+      filteredCountries = this.initTerritories(filteredTerritories, filteredCountries);
+    }
```
Seems I previously incorrectly suggested to place regions filter above territories initialization. Thanks for proper placement!
